jvmti-access/src/main/jni/VirtualThreadSupport.cpp

```diff
@@ -0,0 +1,214 @@
+#include "VirtualThreadSupport.h"
+#include <cstring>
+
+namespace elastic {
+    namespace jvmti_agent {
+
+        namespace {
+
+            bool isExpectedMountEvent(jvmtiExtensionEventInfo& eventInfo) {
```

Comment:
[minor] those two methods seem very close or identical, would there be benefits to merging them in a single one ?

Reply:
I think it's a case of incidental duplication: This methods check for the signature of two different events (unmount and mount) which are not guaranteed to evolve together, so it is better to keep the duplication here imo.